File: exercises/beecrowd/1164/1164.ts
// https://judge.beecrowd.com/pt/problems/view/1164
import * as fs from "fs";

function main(): void {
  const lines: number[] = fs
    .readFileSync("/dev/stdin", "utf8")
    .trim()
    .split("\n")
    .map(Number);

  let lineIndex: number = 0;
  const numTestCases: number = lines[lineIndex++];

  for (let i: number = 0; i < numTestCases; i++) {
    const num: number = lines[lineIndex++];

    if (num <= 1) {
      console.log(`${num} nao eh perfeito`);
      continue;
    }

    let sumDivisors: number = 1;

    const sqrtNum: number = Math.sqrt(num);
    for (let j: number = 2; j <= sqrtNum; j++) {
      if (num % j === 0) {
        sumDivisors += j;
        if (j * j !== num) {
          sumDivisors += num / j;
        }
      }
    }

    if (sumDivisors === num) {
      console.log(`${num} eh perfeito`);
    } else {
      console.log(`${num} nao eh perfeito`);
    }
  }
}

main();

File: exercises/beecrowd/1164/1164.go
// https://judge.beecrowd.com/pt/problems/view/1164
package main

import (
	"fmt"
	"math"
)

func main() {
	var nums int
	fmt.Scan(&nums)

	for i := 0; i < nums; i++ {
		var num int
		fmt.Scan(&num)

		if num <= 1 {
			fmt.Printf("%d nao eh perfeito\n", num)
			continue
		}

		sumDivisors := 1

		for j := 2; j <= int(math.Sqrt(float64(num))); j++ {
			if num%j == 0 {
				sumDivisors += j
				if j*j != num {
					sumDivisors += num / j
				}
			}
		}

		if sumDivisors == num {
			fmt.Printf("%d eh perfeito\n", num)
		} else {
			fmt.Printf("%d nao eh perfeito\n", num)
		}
	}
}
